### ISSUE TITLE
False positive unnecessary-ellipsis for Protocol method inside TYPE_CHECKING

### ISSUE THREAD

**Issue by adamtheturtle** [pylint-dev/pylint#11459]
### Bug description

[Pylint #9319](https://github.com/pylint-dev/pylint/issues/9319) fixed `unnecessary-ellipsis` for ordinary `Protocol` methods in 4.0.9. The warning remains when the method is inside an `if TYPE_CHECKING:` block in the same `Protocol`.

Save this as `protocol_type_checking.py`:

```python
"""Minimal protocol example."""
from typing import TYPE_CHECKING, Protocol

class Example(Protocol):
    """Example protocol."""
    def ordinary(self) -> int:
        """An ordinary protocol method."""
        ...

    if TYPE_CHECKING:
        def static_only(self) -> int:
            """A static-only protocol method."""
            ...
```

The ellipsis in `ordinary` is accepted; the one in `static_only` produces W2301. This occurs in [Literalizer's `Language` protocol](https://github.com/adamtheturtle/literalizer/blob/main/src/literalizer/_language.py), where three static-only methods currently use `raise NotImplementedError` to avoid the warning.

### Configuration

No configuration is needed.

### Command used

```shell
uvx --from pylint==4.0.9 pylint --disable=all --enable=unnecessary-ellipsis protocol_type_checking.py
```

### Pylint output

```text
************* Module protocol_type_checking
protocol_type_checking.py:13:12: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)

-----------------------------------
Your code has been rated at 8.57/10
```

### Expected behavior

No `unnecessary-ellipsis` warning for either `Protocol` method. Both ellipses represent unimplemented method bodies.

### Pylint version

```text
pylint 4.0.9
astroid 4.0.4
Python 3.14.4 (main, Apr 14 2026, 14:46:33) [Clang 22.1.3 ]
```

### OS / Environment

macOS.

### Additional dependencies

None.
